kafka/consumer/subscription_state.py

```diff
@@ -120,6 +124,32 @@ def subscribe(self, topics=(), pattern=None, listener=None):
             raise TypeError('listener must be a ConsumerRebalanceListener')
         self.listener = listener
 
+    def _ensure_valid_topic_name(self, topic):
+        """ Ensures that the topic name is valid according to the kafka source.
+
+        Raises:
+            TypeError: if the topic is
```

Comment:
nit: now that you're raising `ValueError`, this docstring should also be updated. Either clarify what you raise, or just say that you raise on X condidtions, without specifying what you raise (I'm fine with either given that this is a somewhat buried docstring, other maintainers may have other opinions).

Reply:
I tried to adapt to the style of the stuff of the project that I removed ([see this](https://github.com/dpkp/kafka-python/pull/1238/files/09db2238bad317717916941b68e75ff196d43134#diff-fa1fdd480849c0108501eb0ee33b6123L131)).

Now, the private method raises something. Shouldn't this be documented in the public method that uses this method (since it uses it only for that)?

Reply:
I usually only add a docstring to a private method if it isn't obvious from the code what or why it is doing. The audience for private methods is primarily developers with access to the source code. But the public methods end up getting pulled out into HTML docs on kafka-python.readthedocs.org , so those are much more important.

Reply:
@dpkp right now the `SubscriptionState` docs are not visible on http://kafka-python.readthedocs.io/en/master/apidoc/modules.html, do you want to change that?

I'm fine with dropping the docstring from the private method, it reduces the risk of the docstring going out of sync with the code.